fda/functional_data.py

```diff
@@ -223,13 +223,17 @@ def _evaluate_grid(self, axes, *, derivative=0, extrapolation=None,
         return res.reshape(shape)
 
 
-    def _join_evaluation(self, index, index_ext, index_ev, res_extrapolation,
-                         res_evaluation):
+    def _join_evaluation(self, index_matrix, index_ext, index_ev,
+                         res_extrapolation, res_evaluation):
         """Join the points evaluated using evaluation and by the direct
         evaluation.
 
         Args:
             index (ndarray): Boolean index with the points extrapolated.
```

Comment:
Change the comment to reflect the new name.

Reply:
Done.